crates/router/src/core/payments/helpers.rs

```diff
@@ -1777,29 +1780,84 @@ pub async fn retrieve_card_with_permanent_token(
         .change_context(errors::ApiErrorResponse::UnprocessableEntity {
             message: "no customer id provided for the payment".to_string(),
         })?;
+
+    let db = state.store.as_ref();
+    let key_manager_state = &state.into();
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(
+            key_manager_state,
+            &payment_intent.merchant_id,
+            merchant_key_store,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
Can't we pass merchant account to the function instead of fetching it here?

Reply:
this will be moved to configs from merchant account table.